model/registry/deployment.yaml

```diff
@@ -5,18 +5,45 @@ groups:
     brief: >
      This document defines attributes for software deployments.
     attributes:
-      - id: environment
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the deployment.
+        examples: ['deploy my app', 'deploy-frontend']
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          The id of the deployment.
+        examples: ['1208']
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: failed
+              value: failed
+              brief: failed
+              stability: experimental
+            - id: succeeded
```

Comment:
That's an interesting point. For DORA metrics, we want to understand both success & failures. Currently I'd see this as an event attached to a span. Several current CICD systems differentiate between failures and successes as it relates to deployments. This matches to the test case* statuses too.

Reply:
could you use `error.type` attribute then? 

It could be used on spans, events and metrics. 
the lack of it means success, the presence of it means failure, the value describes a class of errors.

This would align with other metrics we have in otel

Reply:
As for test case status, there are more statuses there + test status matches something very popular and pre-existing. 

Deployment status seems to be more of an abstract concept.

Reply:
Documenting this here as a followup to the WG conversation on 07/08/24. 

We discussed this verbally and decided to keep deployment status around due to there being prior art for it. 

* GitHub [deployment events](https://docs.github.com/en/webhooks/webhook-events-and-payloads#deployment_status) contain a deployment status which signifies if the deployment was successful or failed. 
* GitLab [deployment events](https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#deployment-events) also contain a deployment status.
* ArgoCD contains a [on-deployed](https://argocd-notifications.readthedocs.io/en/stable/catalog/#app-deployed) trigger which is similar to deployment status and can be used by the notifications engine to trigger a notification of a deployment. Evaluates app health + sync status.

This does slightly differ from CDEvents. CDEvents uses words like `deployed`, `created` instead of status = success on a deployment. 

We can re-evaluate again in the future if necessary. For now, opting for a status value.